Core/Src/Tasks/ContactorGatekeeperTask.cpp

```diff
@@ -0,0 +1,127 @@
+#include "ContactorGatekeeperTask.h"
+#include<list>
+
+void contactorGatekeeperTask(void* arg)
+{
+    for (;;)
+    {
+        contactorGatekeeper();
+    }
+}
+
+Contactor common = {
+    COMMON_SENSE_GPIO_Port, 
+    COMMON_SENSE_Pin,
+    COMMON_CURRENT_CHANNEL,
+    COMMON_ENABLE_GPIO_Port, 
+    COMMON_ENABLE_Pin,
+    COMMON_CLOSED;
+    COMMON_OPENED
+};
+
+Contactor charge = {
+    CHARGE_SENSE_GPIO_Port, 
+    CHARGE_SENSE_Pin,
+    CHARGE_CURRENT_CHANNEL,
+    CHARGE_ENABLE_GPIO_Port, 
+    CHARGE_ENABLE_Pin,
+    CHARGE_CLOSED;
+    CHARGE_OPENED
+
+}; 
+
+Contactor discharge = {
+    DISCHARGE_SENSE_GPIO_Port, 
+    DISCHARGE_SENSE_Pin,
+    DISCHARGE_CURRENT_CHANNEL,
+    DISCHARGE_ENABLE_GPIO_Port, 
+    DISCHARGE_ENABLE_Pin,
+    DISCHARGE_CLOSED;
+    DISCHARGE_OPENED
+
+}; 
+
+std::unordered_map<uint8_t, Contactor> contactorMap = {{0, common}, {1, charge}, {2, discharge}};
+//added this below for the list of the dischargeStates, idk if this right
+std::list<Contactor> isContactorClosed = {};
+//should switch to ETL if we run into any errors!!!!
+
+
+//Ask jessie why this is in C++
+
+
+void closeContactor(uint8_t contactorToClose)
+{
+
+    // read sense pin
+    Contactor contactorInfo = contactorMap[contactorToClose];
+    //set charge state to closing
+    auxBmsContactorState.contactorInfo = CLOSING;
+    //Enable the contactor then wait
+    uint8_t orionDischargeEnableSense = HAL_GPIO_ReadPin(discharge.sensePort, discharge.sensePin);
+    uint8_t orionChargeEnableSense = HAL_GPIO_ReadPin(charge.sensePort, charge.sensePin);
+    uint8_t orionCommonEnableSense = HAL_GPIO_ReadPin(common.sensePort, common.sensePin);
```

Comment:
Why is this orion? I thought we aren't going to have an "orion" rather it be battery control? I think we could consider renaming this in general.